safe_transaction_service/history/models.py

```diff
@@ -1283,13 +1284,8 @@ def __str__(self):
             return f"{self.safe} - {self.to} - 0x{bytes(self.data).hex()[:6]}"
 
     @property
-    def execution_date(self) -> Optional[datetime.datetime]:
-        if (
-            self.internal_tx.ethereum_tx_id
-            and self.internal_tx.ethereum_tx.block_id is not None
-        ):
-            return self.internal_tx.ethereum_tx.block.timestamp
-        return None
+    def execution_date(self) -> datetime.datetime:
+        return self.internal_tx.timestamp
```

Comment:
This field was optional before. Do we have guarantees that we will always have a value here?

Reply:
Yes, as `traces` are always executed and never pending. I don't know why it was optional before

Reply:
Got it 👌